internal/handlers/common/scale.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// GetScaleParam validates the given scale value for collStats and dbStats command.
+//
+// If the value is valid, it returns its int32 representation,
+// otherwise it returns a command error with the given command being mentioned.
+func GetScaleParam(command string, value any) (int32, error) {
+	scaleValue, err := GetWholeNumberParam(value)
+
+	if err == nil {
+		if scaleValue <= 0 {
+			return 0, commonerrors.NewCommandError(
+				commonerrors.ErrValueNegative,
+				fmt.Errorf("BSON field 'scale' value must be >= 1, actual value '%d'", scaleValue),
+			)
+		}
+
+		if scaleValue > math.MaxInt32 {
+			return math.MaxInt32, nil
+		}
+
+		return int32(scaleValue), nil
+	}
+
+	switch {
+	case errors.Is(err, errUnexpectedType):
+		if _, ok := value.(types.NullType); ok {
+			return 1, nil
+		}
+
+		return 0, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			fmt.Sprintf(
+				`BSON field '%s.scale' is the wrong type '%s', expected types '[long, int, decimal, double]'`,
```

Comment:
So `command` can actually be not command, it can be `$collStats (stage)`. Which would make the formatting of `%s.scale` weird. I suggest passing `command` and another parameter something like `argument` or better name to store a value like `$collStats (stage)`.

Reply:
Mongo itself returns something like `$collStats.storageStats.scale` for this case, so let's do the same.